eng/packages/http-client-csharp/generator/Azure.Generator/src/AzureClientPlugin.cs

```diff
@@ -30,6 +31,9 @@ public class AzureClientPlugin : ScmCodeModelPlugin
     public override AzureOutputLibrary OutputLibrary => _azureOutputLibrary ??= new();
 
     internal ResourceDetection ResourceDetection { get; } = new();
+    internal ParentDetection ParentDetection { get; } = new();
```

Comment:
Isn't your parent just a property on the client from typespec?

Reply:
As offline discussed, we need to clarify the contract of input types containing the resource hierarchy. And it should be an extensible way of holding it in Azure plugin only.
Then, we can put the temp workaround of resource detection close to the input instead of output.

Reply:
Isolated Resource related logic in ResourceBuilder, so we have separated the temp workaround to a central place.

Reply:
I think we will need to have resource related logic in MPG emitter, such as call `getArmResources` API to get the resource hierarchy. MGC doesn't need to provide any extension for this. MPG can have additional input models to hold them.

Reply:
If this is only needed for swagger input, have we considered having this in autorest.csharp only?

We can make it part of the codemodel -> input type conversion before we serialize out to the new plugin.  That way the new plugin only has 1 way to do things which is to read the information directly from the codemodel.json file and it doesn't care how the data got there.